test/fixtures/azure-fixture/src/main/java/fixture/azure/MockAzureBlobStore.java

```diff
@@ -0,0 +1,487 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package fixture.azure;
+
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.bytes.CompositeBytesReference;
+import org.elasticsearch.common.util.Maps;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.rest.RestStatus;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class MockAzureBlobStore {
+
+    private static final Logger logger = LogManager.getLogger(MockAzureBlobStore.class);
+    private static final BytesReference EMPTY = new BytesArray(new byte[] {});
+
+    public enum BlobType {
+        BLOCK("BlockBlob") {
+            @Override
+            AzureBlob create() {
+                return new MockAzureBlockBlob();
+            }
+        },
+        PAGE("PageBlob") {
+            @Override
+            AzureBlob create() {
+                throw new UnsupportedOperationException("PageBlob is not supported");
+            }
+        },
+        APPEND("AppendBlob") {
+            @Override
+            AzureBlob create() {
+                throw new UnsupportedOperationException("AppendBlob is not supported");
+            }
+        };
```

Comment:
I left the existence of other blob types in the code, but we don't implement any of them. These are just stubs that we can fill in as needed.

Reply:
At least this way we should get a more explicit error when someone tries to use one of them

Reply:
I'd rather we made it an immediate test failure (i.e. `ExceptionsHelper.maybeDieOnAnotherThread(new AssertionError("..."))`) rather than having dummy objects that we don't expect ever to use and which explode with a UOE when touched.

Reply:
Addressed in f3351aa60b6c3b03538616f5323630726cbdfd6f